core/src/main/java/io/aiven/klaw/service/TopicOverviewService.java

```diff
@@ -0,0 +1,529 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import io.aiven.klaw.config.ManageDatabase;
+import io.aiven.klaw.dao.Acl;
+import io.aiven.klaw.dao.Env;
+import io.aiven.klaw.dao.KwClusters;
+import io.aiven.klaw.dao.Topic;
+import io.aiven.klaw.helpers.HandleDbRequests;
+import io.aiven.klaw.model.AclInfo;
+import io.aiven.klaw.model.ApiResultStatus;
+import io.aiven.klaw.model.KafkaClustersType;
+import io.aiven.klaw.model.TopicHistory;
+import io.aiven.klaw.model.TopicInfo;
+import io.aiven.klaw.model.TopicOverview;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+public class TopicOverviewService {
+
+  public static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+  public static final ObjectWriter WRITER_WITH_DEFAULT_PRETTY_PRINTER =
+      OBJECT_MAPPER.writerWithDefaultPrettyPrinter();
+  @Autowired ManageDatabase manageDatabase;
+  @Autowired ClusterApiService clusterApiService;
+  @Autowired private CommonUtilsService commonUtilsService;
+  private final MailUtils mailService;
+
+  public TopicOverviewService(MailUtils mailService) {
+    this.mailService = mailService;
+  }
+
+  public TopicOverview getTopicOverview(String topicNameSearch) {
+    log.debug("getAcls {}", topicNameSearch);
+    String userDetails = getUserName();
+    HandleDbRequests handleDb = manageDatabase.getHandleDbRequests();
+    int tenantId = commonUtilsService.getTenantId(getUserName());
+
+    if (topicNameSearch != null) {
+      topicNameSearch = topicNameSearch.trim();
+    } else {
+      return null;
+    }
+
+    Integer loggedInUserTeam = getMyTeamId(userDetails);
+    List<Topic> topics = handleDb.getTopics(topicNameSearch, tenantId);
+
+    // tenant filtering
+    List<String> allowedEnvIdList = getEnvsFromUserId(userDetails);
+    topics =
+        topics.stream()
+            .filter(topicObj -> allowedEnvIdList.contains(topicObj.getEnvironment()))
+            .collect(Collectors.toList());
+
+    TopicOverview topicOverview = new TopicOverview();
+
+    if (topics.size() == 0) {
+      topicOverview.setTopicExists(false);
+      return topicOverview;
+    } else {
+      topicOverview.setTopicExists(true);
+    }
+
+    String syncCluster;
+    String[] reqTopicsEnvs;
+    ArrayList<String> reqTopicsEnvsList = new ArrayList<>();
+    try {
+      syncCluster = manageDatabase.getTenantConfig().get(tenantId).getBaseSyncEnvironment();
+    } catch (Exception exception) {
+      syncCluster = null;
```

Comment:
Since here could be any exception it is worth logging it because it could also bugs appear here

Reply:
Agree, logged.